e2e/helpers/tableColumnSchemaEditor.ts

```diff
@@ -0,0 +1,294 @@
+    await columnMaxListLengthFields
+      .nth(columnIndex)
+      .fill(maxListLength.toString())
+  })
+}
+
+const setColumnDefaultValue = async (
+  page: Page,
+  columnIndex: number,
+  defaultValue: string,
+  columnType: string,
+) => {
+  await test.step('set column default value', async () => {
+    const columnDefaultValueField = page
+      .getByLabel('Default Value')
+      .nth(columnIndex)
+    if (columnType === 'Boolean') {
+      await columnDefaultValueField.click()
+      await page.getByRole('option', { name: defaultValue }).click()
+    } else {
+      await columnDefaultValueField.fill(defaultValue)
+    }
+  })
+}
+
+const enterColumnRestrictValues = async (
+  page: Page,
+  columnIndex: number,
+  values: string[],
+) => {
+  await test.step('enter restrict values', async () => {
+    await page.getByLabel('Restrict Values').nth(columnIndex).click()
+
+    const editValuesHeader = page.getByText('Edit Values')
+    await expect(editValuesHeader).toBeVisible()
+
+    await page.getByRole('button', { name: 'Paste new values' }).click()
+    await page
+      .getByPlaceholder('Place comma or tab delimited values here')
+      .fill(values.join(','))
+    await page.getByRole('button', { name: 'Add', exact: true }).click()
+    await page.getByRole('button', { name: 'OK' }).click()
+
+    await expect(editValuesHeader).not.toBeVisible()
+  })
+}
+
+const selectColumnFacet = async (
+  page: Page,
+  columnIndex: number,
+  facetType: FacetType,
+) => {
+  await test.step('set column facet', async () => {
+    const columnFacetFields = page.getByLabel('Facet Type')
+    await columnFacetFields.nth(columnIndex).click()
+    await page.getByRole('option', { name: facetType }).click()
+  })
+}
+
+// Set Table Schema ------------------------------------------------------------
+const enterColumnSchema = async (
+  page: Page,
+  columnConfig: ColumnSchemaConfig,
+  columnIndex: number,
+) => {
+  if (columnConfig.name) {
+    await enterColumnName(page, columnIndex, columnConfig.name)
+  }
+  if (columnConfig.type) {
+    await selectColumnType(page, columnIndex, columnConfig.type)
+  }
+  if (columnConfig.size) {
+    await setColumnMaxSize(page, columnIndex, columnConfig.size)
+  }
+  if (columnConfig.maxListLength) {
+    await setColumnMaxListLength(page, columnIndex, columnConfig.maxListLength)
+  }
+  if (columnConfig.defaultValue) {
+    await setColumnDefaultValue(
+      page,
+      columnIndex,
+      columnConfig.defaultValue,
+      columnConfig.type,
+    )
+  }
+  if (columnConfig.restrictValues) {
+    await enterColumnRestrictValues(
+      page,
+      columnIndex,
+      columnConfig.restrictValues,
+    )
+  }
+  if (columnConfig.facet) {
+    await selectColumnFacet(page, columnIndex, columnConfig.facet)
+  }
+}
+
+export const addColumnsFromColumnSchemaConfig = async (
+  page: Page,
+  columnConfigs: ColumnSchemaConfig[],
+  columnStartingIndex: number = 0,
```

Comment:
By default, enter all of the columns in the provided config into the schema editor. However, optionally specify the config index at which to start entering columns in the schema editor. Allows this function to be reused when updating an existing column schema.

Reply:
Doesn't it always click the "Add column" button? If you edited an existing column, you'd end up with an extra empty column at the end, right?

Reply:
Oh good call, that's right. I'll change the function to calculate the current number of columns, so the starting index doesn't have to be passed. If we need to edit the schema, we can add an "editSchema..." function. Thanks!